File: js/tests/randomness.test.ts
import style, { Color, Style } from "../src/index";

describe("Randomness Test: Styled Terminal Output", () => {
  const TIMES = 200;

  it(`prints ${TIMES} random colors to check randomness.`, () => {
    const history = new Set();
    const text = "   ";

    for (let i = 0; i < TIMES; ++i) {
      const v = style.bg(Color.random).apply(text);
      history.add(v);
    }
    console.log([...history].join(""));

    expect(history.size).toBe(TIMES);
  });
});

File: src/py/tests/randomness.test.py
from styled_terminal import style, Color


# Function to be tested
def print_randomness():
    for _ in range(500):
        print(style.bg(Color.random()).apply("  "), end="")


# If this file is run directly, run the tests
if __name__ == "__main__":
    print_randomness()
